samples/apps/copilot-chat-app/webapi/DeploymentTemplates/sk-existing-ai.json

```diff
@@ -0,0 +1,331 @@
+  "contentVersion": "1.0.0.0",
+  "metadata": {
+    "_generator": {
+      "name": "bicep",
+      "version": "0.16.2.56959",
+      "templateHash": "5205015569073245261"
+    }
+  },
+  "parameters": {
+    "name": {
+      "type": "string",
+      "defaultValue": "sk",
+      "metadata": {
+        "description": "Name for the deployment"
+      }
+    },
+    "appServiceSku": {
+      "type": "string",
+      "defaultValue": "B1",
+      "metadata": {
+        "description": "SKU for the Azure App Service plan"
+      }
+    },
+    "packageUri": {
+      "type": "string",
+      "defaultValue": "https://skaasdeploy.blob.core.windows.net/api/skaas.zip",
+      "metadata": {
+        "description": "Location of package to deploy as the web service"
+      }
+    },
+    "aiService": {
+      "type": "string",
+      "defaultValue": "AzureOpenAI",
+      "allowedValues": [
+        "AzureOpenAI",
+        "OpenAI"
+      ],
+      "metadata": {
+        "description": "Underlying AI service"
+      }
+    },
+    "completionModel": {
+      "type": "string",
+      "defaultValue": "gpt-35-turbo",
+      "metadata": {
+        "description": "Model to use for chat completions"
+      }
+    },
+    "embeddingModel": {
+      "type": "string",
+      "defaultValue": "text-embedding-ada-002",
+      "metadata": {
+        "description": "Model to use for text embeddings"
+      }
+    },
+    "endpoint": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Azure OpenAI endpoint to use (ignored when AI service is not AzureOpenAI)"
+      }
+    },
+    "apiKey": {
+      "type": "securestring",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Azure OpenAI or OpenAI API key"
+      }
+    }
+  },
+  "variables": {
+    "location": "[resourceGroup().location]",
+    "uniqueName": "[format('{0}-{1}', parameters('name'), uniqueString(resourceGroup().id))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Web/serverfarms",
+      "apiVersion": "2022-03-01",
+      "name": "[format('{0}-plan', variables('uniqueName'))]",
+      "location": "[variables('location')]",
+      "sku": {
+        "name": "[parameters('appServiceSku')]"
+      }
+    },
+    {
+      "type": "Microsoft.Web/sites",
+      "apiVersion": "2022-03-01",
+      "name": "[format('{0}sk-web', variables('uniqueName'))]",
+      "location": "[variables('location')]",
+      "tags": {
+        "skweb": "1"
+      },
+      "properties": {
+        "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', format('{0}-plan', variables('uniqueName')))]",
+        "httpsOnly": true,
+        "siteConfig": {
+          "alwaysOn": true,
+          "detailedErrorLoggingEnabled": true,
+          "minTlsVersion": "1.2",
+          "netFrameworkVersion": "v6.0",
+          "use32BitWorkerProcess": false,
+          "appSettings": [
```

Comment:
This looks like anytime we update appsettings we'll need to update this as well. Could we just have the service use the appsettings that is deployed and just use this for overrides?

Reply:
Yes. Removed a bunch of those but kept the ones needed